src/riot-rs-embassy/src/arch/dummy/mod.rs

```diff
@@ -28,6 +28,8 @@ impl From<Peripherals> for OptionalPeripherals {
 #[derive(Default)]
 pub struct Config;
 
+#[allow(clippy::needless_pass_by_value, reason = "dummy implementation")]
+#[must_use]
```

Comment:
What do you think about adding `clippy::must_use_candidate` to the allowed list in the workspace? 
- according to the lint description: "_this lint just shows places where you could add the attribute._"  and it has many false positives[^1]
- the `#[must_use]` annotation creates a lot of noise in the code and makes it difficult to distinguish the cases where it is actually important that a value is used

[^1]: https://rust-lang.github.io/rust-clippy/master/index.html#/must_use_candidate